pkg/daemon/winsvc/winsvc.go

```diff
@@ -28,11 +30,27 @@ func EnsureServiceState(service Service, state svc.State) error {
 	}
 	switch state {
 	case svc.Running:
-		return service.Start()
+		err = service.Start()
+		if err != nil {
+			return err
+		}
 	case svc.Stopped:
 		_, err = service.Control(svc.Stop)
-		return err
+		if err != nil {
+			return err
+		}
 	default:
 		return errors.New("unexpected state request")
 	}
+	// wait for the state change to actually take place
+	return wait.PollImmediate(retry.Interval, retry.ResourceChangeTimeout, func() (bool, error) {
```

Comment:
Nit: you should update the doc for `retry.ResourceChangeTimeout`.

Reply:
I think it's still applicable: 

ResourceChangeTimeout is the total time waited for a change (create/**update**/delete) to take place

Reply:
I guess `Resource` is generic enough.